openshift/machine-config-operator
# MCO-552: implement the ability for the MCO to handle image registry certificates

**- What I did**

The MCO can now manage image registry certs

1) the MCO reads from image.config.openshift.io/cluster and looks for additionalTrustedCA, which is a user specified additional registry

2) the MCO reads from openshift-config-managed/image-registry-ca configmap which is to be created and managed by the image registry operator

3) the MCO creates and writes to an additional openshift-config-managed/merged-trusted-image-registry-cas configmap with the combined data

4) we write this all to disk in the daemon as we now do with other certificates. These paths follow: /etc/docker/certs.d/<CONFIGMAP_KEY>/ca.crt format

**- How to verify it**

oc describe controllerconfig to see the data from the CMs

ssh into the node and check out /etc/docker/certs.d to see if the files propagate.

**- Description for the changelog**
 
The MCO now supports managing image registry certificates and writes them to disk.
